packages/data-context/src/data/ProjectConfigIpc.ts

```diff
@@ -324,6 +326,8 @@ export class ProjectConfigIpc extends EventEmitter {
         } else {
           childOptions.env.NODE_OPTIONS = tsNodeLoader
         }
+
+        childOptions.env['NODE_OPTIONS'] = `--require ${this.projectRoot}/.pnp.cjs ${ childOptions.env['NODE_OPTIONS']}`
```

Comment:
This line alone fixes 22747, but only if webpack-dev-server is not used (e.g. global mode and testing an external angular project run via ng serve)